app/auth/middleware.test.js

```diff
@@ -46,8 +43,13 @@ describe('Authentication middleware', function() {
     })
 
     context('when there is no grant response', function() {
-      beforeEach(function() {
-        authentication.processAuthResponse()(req, {}, nextSpy)
+      beforeEach(async function() {
+        const authentication = proxyquire('./middleware', {
```

Comment:
You can leave the proxyquire definition at the top this where it was and pass a `sinon.stub()` into the `getUserLocations` method. Then use `. rejects()` and `.resolves()` in the test before functions.

At top of file:
```js
const getUserLocationsStub = sinon.stub()
const authentication = proxyquire('./middleware', {
  '../../common/lib/user': UserStub,
  '../../common/services/user-locations': {
    getUserLocations: getUserLocationsStub,
  },
})
```

In success `beforeEach`:

```js
getUserLocationsStub.resolves(['TEST'])
```

In failure `beforeEach`:

```js
getUserLocationsStub.rejects(userLocationFailureError)
```

Reply:
Ah, clever thanks. I took this implementation from another test.

Reply:
I've had a go at this, but I don't think the dynamic changing of the stub works in this way. It seems that the behaviour is fixed once `proxyquire` loads it. I think that's why it was also implemented this way in https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/app/healthcheck/middleware.test.js. Happy for you to have a crack at it.